internal/controllers/clusterextension_controller.go

```diff
@@ -409,6 +411,17 @@ func (r *ClusterExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&ocv1alpha1.ClusterExtension{}).
 		Watches(&catalogd.Catalog{},
 			handler.EnqueueRequestsFromMapFunc(clusterExtensionRequestsForCatalog(mgr.GetClient(), mgr.GetLogger()))).
+		WithEventFilter(predicate.Funcs{
+			UpdateFunc: func(ue event.UpdateEvent) bool {
+				oldObject, isOldCatalog := ue.ObjectOld.(*catalogd.Catalog)
+				newObject, isNewCatalog := ue.ObjectNew.(*catalogd.Catalog)
+
+				if !isOldCatalog || !isNewCatalog {
+					return true
+				}
+				return oldObject.Status.ResolvedSource.Image.Ref != newObject.Status.ResolvedSource.Image.Ref
```

Comment:
`.Status.ResolvedSource` and `.Status.ResolveSource.Image` are both pointers.

I assume we need to handle cases where either of those is `nil` for both the old and new objects?